configs/provider_requirements.go

```diff
@@ -7,14 +7,20 @@ import (
 )
 
 // RequiredProvider represents a declaration of a dependency on a particular
-// provider version without actually configuring that provider. This is used in
-// child modules that expect a provider to be passed in from their parent.
+// provider version or source without actually configuring that provider. This
+// is used in child modules that expect a provider to be passed in from their
+// parent.
 type RequiredProvider struct {
 	Name        string
-	Source      string // TODO
+	Source      Source
```

Comment:
Here is one notable non-test change: I added a `DeclRange` to the provider source struct for nicer  user-facing errors and warnings

Reply:
Me too! Source range all the things, I think.